plugin-server/src/main/ingestion-queues/analytics-events-ingestion-consumer.ts

```diff
@@ -78,50 +78,67 @@ export async function eachBatchIngestionWithOverflow(
         const batches = []
         const seenIds: Set<string> = new Set()
         let currentBatch: KafkaMessage[] = []
+
         for (const message of kafkaMessages) {
             const pluginEvent = formPipelineEvent(message)
             const seenKey = `${pluginEvent.team_id}:${pluginEvent.distinct_id}`
+
+            // Events with a null key should have been produced to the the
+            // KAFKA_EVENTS_PLUGIN_INGESTION_OVERFLOWtopic, so we shouldn't see them here as this consumer's
+            // topic is set to KAFKA_EVENTS_PLUGIN_INGESTION. However, there could be some lingering events
+            // from before the new *_OVERFLOW topic was initialized. Any events with a null key or that
+            // exceed capacity are redirected to the *_OVERFLOW topic.
+            if (message.key == null || ConfiguredLimiter.consume(seenKey, 1) === false) {
+                // Set message key to be null so we know to send it to overflow topic.
+                // We don't want to do it here to preserve the kafka offset handling
+                message.key = null
+                currentBatch.push(message)
```

Comment:
note that if we do it this way, we're not respecting batch sizes & we should. 
This is worse given below we use `currentBatch.length === batchSize` not `currentBatch.length >= batchSize` (lets change that too for safety in the future), so we'd potentially end up not breaking into batches at all.

Reply:
Gave it a few minutes of thought, there is a solution here with `slice`s, but I'm getting too smart, so I'm just using an `if`.